### PR TITLE
gtfTools filter codingOnly

### PR DESCRIPTION
**Pull request by RavenYL**
# Description
- Added an option to gtfTools filter to remove transcripts that are non-coding, i.e. do not have any CDS

### Testing
- Added a test for the option. Took lines from hs1 ncbiRefSeq gtf.

### Checklist before requesting a review
- [x] I performed a self-review of my code
- [x] If it this a core feature, I added thorough tests
- [x] The command `go fmt` or `make clean` was used on all files included

### CODE REVIEW

**Review by cfauci — APPROVED** [review]
Definitely much needed!

File: cmd/gtfTools/filter.go [hunk 1]
```diff
@@ -71,8 +74,29 @@ func gtfFilter(s FilterSettings) {
 	}
 
 	records := gtf.Read(s.InFile)
-	for currGene := range records {
+	for currGene := range records { // for each gene
 		pass = true
+
+		if s.CodingTranscript { // option to detect coding transcript
+			for _, currTranscript := range records[currGene].Transcripts { // for each transcript
+				if len(currTranscript.Exons) == 0 { // if transcript has no exon, then transcript is non-coding
```
Comment (RavenYL) [inline cmd/gtfTools/filter.go:82]: Craig's comment 1: now the pass is for gene, but decision to pass is based on gene's last transcript. 
Christi's suggestion: Set pass to false when go into transcript. Only pass transcripts that are coding.
Craig's comment 2: Line 82 if may not be necessary.

**Review by cfauci — DISMISSED** [review]

File: cmd/gtfTools/filter.go [hunk 2]
```diff
@@ -71,8 +74,27 @@ func gtfFilter(s FilterSettings) {
 	}
 
 	records := gtf.Read(s.InFile)
-	for currGene := range records {
+	for currGene := range records { // for each gene
 		pass = true
+
+		if s.CodingTranscript { // option to detect coding transcript
+			transcriptCoding := false                                      // in each gene, for each transcript, transcript is non-coding unless an exon with CDS is found later
```
Comment (sweavs111) [inline cmd/gtfTools/filter.go:81]: minor memory efficiency: declare variable outside of for loop

File: cmd/gtfTools/filter.go [hunk 3]
```diff
@@ -71,8 +74,27 @@ func gtfFilter(s FilterSettings) {
...
+			for _, currTranscript := range records[currGene].Transcripts { // for each transcript
```
Comment (sweavs111) [inline cmd/gtfTools/filter.go:82]: could declare currTranscript outside of for loop if you wanted

File: cmd/gtfTools/filter.go [hunk 4]
```diff
@@ -71,8 +74,27 @@ func gtfFilter(s FilterSettings) {
...
+				if len(currTranscript.Exons) != 0 { // if transcript has no exon, then transcript is non-coding. if transcript has exon(s),
+					// if we find at least 1 exon that has 1 CDS, then transcript is coding
+					for _, currExon := range currTranscript.Exons { // for each exon
```
Comment (sweavs111) [inline cmd/gtfTools/filter.go:85]: could declare currExon outside of for loop

**Review by sweavs111 — COMMENTED** [review]
looks great! i just suggested some minor efficiency-saves

**Review by sweavs111 — APPROVED** [review]
looks great!

**Review by hcnapier — APPROVED** [review]

**Pull request merged**